### PR TITLE
Update the script of the upgrade tests for serving

### PR DESCRIPTION
**Pull request by houshengbo**
## Proposed Changes

* Serving just updates the script of the upgrade tests. Operator needs to catch up the updates.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -38,7 +38,6 @@ cloud.google.com/go/storage v1.0.0/go.mod h1:IhtSnM/ZTZV8YYJWCY8RULGVqBDmpoyjwiy
 cloud.google.com/go/storage v1.5.0/go.mod h1:tpKbwo567HUNpVclU5sGELwQWBDZ8gh0ZeosJ0Rtdos=
 cloud.google.com/go/storage v1.6.0/go.mod h1:N7U0C8pVQ/+NIKOBQyamJIeKQKkZ+mxpohlUTyfDhBk=
 cloud.google.com/go/storage v1.8.0/go.mod h1:Wv1Oy7z6Yz3DshWRJFhqM/UCfaWIRTdp0RXyy7KQOVs=
-cloud.google.com/go/storage v1.10.0 h1:STgFzyU5/8miMl0//zKh2aQeTyeaUH3WN9bSUiJ09bA=
```
Comment (markusthoemmes) [inline go.sum:41]: These changes seem quite unrelated. Could we drop them?

Comment (houshengbo) [reply go.sum:41]: weird I just run update-codegen.sh

Comment (markusthoemmes) [reply go.sum:41]: might be because you're using a newer Golang version or something.

Comment (houshengbo) [reply go.sum:41]: but check here: https://github.com/knative/operator/pull/566/files#diff-3295df7234525439d778f1b282d146a4f1ff6b415248aaac074e8042d9f42d63L41

This is auto-generated as well.

File: test/e2e-serving-upgrade-tests.sh [hunk 2]
```diff
@@ -69,8 +66,17 @@ function test_setup() {
...
+function create_test_namespace_serving() {
+  kubectl get ns ${SERVING_TESTS_NAMESPACE} || kubectl create namespace ${SERVING_TESTS_NAMESPACE}
+  kubectl get ns ${SERVING_TESTS_ALT_NAMESPACE} || kubectl create namespace ${SERVING_TESTS_ALT_NAMESPACE}
+}
+
 # Skip installing istio as an add-on.
-initialize "$@" --skip-istio-addon
+# Temporarily increasing the cluster size for serving tests to rule out
+# resource/eviction as causes of flakiness.
+# Pin to 1.18 since scale test is super flakey on 1.19
```
Comment (markusthoemmes) [inline test/e2e-serving-upgrade-tests.sh:78]: Do we even run the scale tests here?

Comment (houshengbo) [reply test/e2e-serving-upgrade-tests.sh:78]: We ran the tests including:
```
test/upgrade: TestServingUpgrades/PreUpgradeTests/ServicePreUpgradeAndScaleToZeroTest 
test/upgrade: TestServingUpgrades/PreUpgradeTests/InitialScalePreUpgradeTest
test/upgrade: TestServingUpgrades/PostUpgradeTests/PostUpgradeFromScaleToZeroTest
test/upgrade: TestServingUpgrades/PostUpgradeTests/InitialScalePostUpgradeTest
```
They are all indicated with `scale`.

Comment (markusthoemmes) [reply test/e2e-serving-upgrade-tests.sh:78]: Those are not the scale tests that are causing trouble in Serving though. They'd be in the scale package, which I believe we're not running here.

Comment (houshengbo) [reply test/e2e-serving-upgrade-tests.sh:78]: then, it is not running the scale tests.

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

File: test/e2e-serving-upgrade-tests.sh [hunk 3]
```diff
@@ -69,6 +66,12 @@ function test_setup() {
   cd ${OPERATOR_DIR}
 }
 
+# Create test namespaces for serving
+function create_test_namespace_serving() {
+  kubectl get ns ${SERVING_TESTS_NAMESPACE} || kubectl create namespace ${SERVING_TESTS_NAMESPACE}
+  kubectl get ns ${SERVING_TESTS_ALT_NAMESPACE} || kubectl create namespace ${SERVING_TESTS_ALT_NAMESPACE}
```
Comment (dprotaso) [inline test/e2e-serving-upgrade-tests.sh:72]: These need the correct labels for the configmap validation webhooks to function

https://github.com/knative/serving/blob/main/config/core/webhooks/configmap-validation.yaml#L30-L33
